docs/source/topics/proc_deploying-sample-application-with-odo.adoc

```diff
@@ -0,0 +1,60 @@
+[id="deploying-sample-application-with-odo_{context}"]
+= Deploying a sample application with `odo`
+
+You can use OpenShift Do ([command]`odo`) to create OpenShift projects and applications from the command line.
+This procedure deploys a sample Java application to the OpenShift cluster running in the {prod} virtual machine.
+
+.Prerequisites
+
+* [command]`odo` is installed.
+For more information, see link:{odo-docs-url-installing}[Installing `odo`] in the [command]`odo` documentation.
+* The {prod} virtual machine is running.
+For more information, see <<starting-the-virtual-machine_{context}>>.
+* You must log in through [command]`oc` as the `developer` user.
+For more information, see <<accessing-the-openshift-cluster-with-oc_{context}>>.
+
+.Procedure
+
+To deploy a sample Java application through [command]`odo`, follow these steps:
+
+. Create a project for your application:
++
+[subs="+quotes,attributes"]
+----
+$ odo project create java-sample
+----
+
+. Create a directory for your components:
++
+[subs="+quotes,attributes"]
+----
+$ mkdir java-sample
+$ cd java-sample
+----
+
+. Create a Java component from a sample application on GitHub:
++
+[subs="+quotes,attributes"]
+----
+$ odo create java --git https://github.com/spring-projects/spring-petclinic.git
```

Comment:
I am bit concern about having a sample app as java, it might take more resources than a simple python/nodejs example, I would first try this before merge this PR. When a new user just start the crc he really have very limited resource to play with the cluster and I don't want first app to be crash because of mem restriction we have :(

Reply:
I have tried this. It works with the default CRC settings on 4.3.10.

We also discussed using a Java sample application during one of the CRC docs sync meetings and agreed that Java is more appropriate for the audience.

Reply:
Quarkus? :) We could add an automated tests for these steps which run once in a while and lets us know if something broke, as it would indeed look bad if the example from our doc failed.

Reply:
Alternatively, if we can find the lightest possible Java sample application, that would also be nice. :)

Honestly, though, I don't have strong opinions about whether we should use Java or Python/Node.js personally.

Reply:
> I am bit concern about having a sample app as java, it might take more resources

This was discussed in several docs-sync. Java seems more appropriate with our other offerings. Odo's documentation can serve the other purposes.

Reply:
It's also worth noting that the only components listed as supported by `odo catalog list components` are `java` and `nodejs`. This suggests, in my opinion, that we should use one of these sample applications as the example for this procedure. If the Java application is too heavy for the average CodeReady Containers OpenShift cluster, that only leaves one option.

@code-ready/crc-devel, please let me know what you think -- would you prefer that we use the Java sample, or a Node.js sample? If possible, I'd like to have this documentation merged before the 1.10 release. Thank you!

Reply:
That specific  https://github.com/spring-projects/spring-petclinic.git application is imo too heavy, in that it needs to download a ton of stuff before building. Hopefully there are more lightweight java apps which can be used instead.

Reply:
It seems to be the "official" java sample for odo :( https://docs.openshift.com/online/starter/cli_reference/openshift_developer_cli/using-sample-applications.html#java

Reply:
@robin-owen you're right about nodejs and java support in odo. They're the only two supported component types right now (and with specific versions)

@cfergeau we're actively working on Quarkus support but the work is tied to DevFile work that's also happening in parallel. I'm optimistic of its support landingin  soon! It is being tracked at https://github.com/openshift/odo/issues/2800.

At the moment, I'm not aware of any examples other than the ones mentioned in the docs. I'll ping you folks as soon as I have something lightweight and java. :disappointed:

Reply:
what is the ETA on Quarkus support? If you can't say here, ping me on slack. A rough estimate would be good enough

Reply:
There's no straightforward answer to this but I think it'll be around July for it to be generally available. In experimental mode that works with devfile, it'll land in a week or two. But "experimental mode" is prone to breaking changes.